contribute/e2e/CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+repositories such as openebs/openebs, openebs/mayaserver, openebs/jiva. This part of the continuous integration is built around a dedicated Jenkins server, which upon detecting commits via a polling mechanism, launches VMs using Vagrant and runs the ansible playbooks to setup the environment and execute e2e tests. The openebs/e2e project focuses on this part of the CI.
+
+The below schema illustrates the workflow
+
+![OpenEBS CI Overview Diagram](https://github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E & OPENEBS CI 
+
+Contributions are welcome !!  Before starting, a reading of the project documentation at [ReadTheDocs](http://openebs.readthedocs.io/en/latest/ ) is highly recommended. Please raise a github issue for any questions you may have on the project in general or e2e,CI in particular.
+You can post questions on the OpenEBS slack channel too.
+
+You could contribute to any of the following areas : 
+
+- Submit proposals to add new tests/test categories to the e2e
+- Create new application specification YAMLs 
+- Create ansible playbooks to execute test workflows
+- Submit changes to fix bugs/improve e2e tests 
+- Raise/fix issues on the CI framework 
+- Raise/fix issues on the e2e, CI documentation
+
+## GENERAL CONTRIBUTION GUIDELINES
+
+Here are some general contribution guidelines. 
+
+### Adding new tests/test-categories
+
+To add new tests, please raise an issue describing the objective, pre-requisites, test steps, expected results & cleanup routines.
+In case of proposals to add new test sections/categories, also add a generic description explaining the purpose and value addition
+
+Once accepted, the issue can be assigned to interested contributors who can make relevant changes with a pull request. 
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, as applicable : 
+
+- https://kubernetes.io/docs/concepts/configuration/overview/
+- https://www.mirantis.com/blog/introduction-to-yaml-creating-a-kubernetes-deployment/
+- https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/
+
+These can be placed in here : https://github.com/openebs/openebs/tree/master/k8s/demo
+
+### Create ansible playbooks to execute test workflows
+
+The test playbooks can be placed in e2e/ansible/playbooks/<test-category>. Here are some considerations 
+
+- Each test playbook is recommended to be constructed with the following template.
+
+  The main test playbook, which statically "includes" following files :
+
+  - Test variables YAML
+  - Pre-requisites tasks YAML
+  - Cleanup tasks YAML
+  
+  Auxiliary scripts that are invoked in the test playbook can be placed in the same directory. 
+
+  A sample test playbook can be seen [here](https://github.com/openebs/openebs/tree/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod)

```

Comment:
You can view a sample test playbook [here].

Reply:
Done